File: lab5-2/lib/analyzer.py
class Analyzer

  def __init__(self, polynom)
    self.polynom = polynom
    self.complexity = polynom.indexes.size
    reset

  def shift(value)
    i = self.polynom.indexes.inject(value) {|input, index| input ^ self.state[index]}
    self.state.unshift(i)
    self.state.pop

  def state
    self.state.dup

  def reset
    self.state = Array.new(self.polynom.to_a.size - 1) { 0 }

File: lab6/lib/analyzer.rb
class Analyzer

  attr_reader :complexity

  def initialize(polynom)
    @polynom = polynom
    @complexity = @polynom.indexes.size
    reset
  end

  def shift(value)
    i = @polynom.indexes.inject(value) {|input, index| input ^ @state[index]}
    @state.unshift(i)
    @state.pop
  end

  def state
    @state.dup
  end

  def reset
    @state = Array.new(@polynom.to_a.size - 1) { 0 }
  end

end
